### ISSUE TITLE
[ENH] Changing and Adding features to CNN Classifier

### ISSUE THREAD

**Issue by AurumnPegasus** [alan-turing-institute/sktime#2935]
**Is your feature request related to a problem? Please describe.**
As discussed in the daily-stand up on 05-07-12, I wanted to propose some enhancements / changes to CNN Classifier (and hence to Deep Learning based estimators)

**Describe the solution you'd like**
- [ ] Moving the `convert_y_to_keras` function in `BaseDeepClassifier` into a utils function
- [x] Enable returning the `label_encoder` and `one hot encoder` to the user from `convert_y_to_keras` (it can be useful for the users in case they want to use it to encode other data / or to decode data later on)
- [x] Add option to have multiple linear layers in output layer / different activation function. Also add the `use_bias` parameter to output layer. (simply as a customisation option to users, multiple linear layers are useful to create deeper networks depending on dataset / choice of activation function)
- [x] option to choose optimiser (usually its Adam but just for customisation's sake we can give this as option as well). Also should allow specifying `lr` either within the optimiser or as parameter to the class itself.
- [x] Accept parameter as input in `CNNClassifier` to specify `random seed`
- [ ] Train/fit the model on GPU depending on users choice (maybe take an input parameter for this as well)
- [x] Option to view / return the loss of fit (either as a graph or simply as a list / array ) / A better idea is to add a summary function which can return model summary as well as `fit` history. 

**Additional context**
These are some additional features / changes I wanted to add to the `CNNClasifier`. Any inputs / additional comments are appreciated

**Comment by fkiraly** [comment]
Yes, great additions, great suggestions!

> Also should allow specifying `lr` either within the optimiser or as parameter to the class itself.

I think this one is unnecessary, because you can pass the `lr` to the optimizer, which is an object you would then pass to the classifier class constructor. We do not need two places to pass it?

Or, do you see a good argument why it would be useful to the user to pass `lr` separately? Or am I simply misunderstanding the suggestion?

**Comment by fkiraly** [comment]
> can we keep the defaults the same to maintain reproducibility of results?

Changing defaults would require deprecation, so I would advise against changing defaults.

**Comment by AurumnPegasus** [comment]
Another thing I was planning was to have `validation_split` as a parameter to make it easier for users. It is simply an additional parameter for us in the `model.fit` method, and should not be difficult. Only issue with this is are we going in accordance to other sktime estimators, do they have functionality to automate validation? In which case we should probably either add it for all (simply use sklearn's `train_test_split` or not use here as well)

**Comment by fkiraly** [comment]
reopening as there are still two checkboxes open, we need to decide what to do with them before we close

**Comment by AurumnPegasus** [comment]
@fkiraly about the two open checkboxes:
There is already an issue opened for GPU functionality, which is in progress: https://github.com/alan-turing-institute/sktime/issues/3020
Issue with the first checkbox (related to `convert_y_to_keras` was related to where should we move it. If we do decide to have a common DL class as discussed here: https://github.com/alan-turing-institute/sktime/issues/3190 we can simply move it there. Otherwise we might need to create a utils file for it.

**Comment by fkiraly** [comment]
Ok. Closing then, as two remaining bullet points are tracked in #3020 and #3190.
